Mathlib/Order/RelIso/Set.lean

```diff
@@ -45,66 +45,61 @@ theorem range_eq (e : r ≃r s) : Set.range e = Set.univ :=
 
 /-- The relation embedding from the inherited relation on a subset. -/
-protected def relEmbedding (r : α → α → Prop) (p : Set α) : Subrel r p ↪r r :=
+protected def relEmbedding (r : α → α → Prop) (p : α → Prop) : Subrel r p ↪r r :=
   ⟨Embedding.subtype _, Iff.rfl⟩
 
 @[simp]
 theorem relEmbedding_apply (r : α → α → Prop) (p a) : Subrel.relEmbedding r p a = a.1 :=
   rfl
 
 /-- A set inclusion as a relation embedding. -/
-protected def inclusionEmbedding (r : α → α → Prop) {p q : Set α} (h : p ⊆ q) :
-    Subrel r p ↪r Subrel r q where
+protected def inclusionEmbedding (r : α → α → Prop) {s t : Set α} (h : s ⊆ t) :
+    Subrel r (· ∈ s) ↪r Subrel r (· ∈ t) where
   toFun := Set.inclusion h
   inj' _ _ h := (Set.inclusion_inj _).mp h
   map_rel_iff' := Iff.rfl
 
 @[simp]
-theorem coe_inclusionEmbedding (r : α → α → Prop) {p q : Set α} (h : p ⊆ q) :
-    (Subrel.inclusionEmbedding r h : p → q) = Set.inclusion h :=
+theorem coe_inclusionEmbedding (r : α → α → Prop) {s t : Set α} (h : s ⊆ t) :
+    (Subrel.inclusionEmbedding r h : s → t) = Set.inclusion h :=
   rfl
 
-instance (r : α → α → Prop) [IsWellOrder α r] (p : Set α) : IsWellOrder p (Subrel r p) :=
+instance (r : α → α → Prop) [IsWellOrder α r] (p : α → Prop) : IsWellOrder _ (Subrel r p) :=
   RelEmbedding.isWellOrder (Subrel.relEmbedding r p)
 
--- TODO: this instance is needed as `simp` automatically simplifies `↑{a // p a}` as `{a | p a}`.
---
--- Should `Subrel` be redefined in terms of `p : α → Prop` instead of `p : Set α` to avoid
--- this issue?
-instance (r : α → α → Prop) (p : α → Prop) [IsWellOrder α r] :
-    IsWellOrder {a // p a} (Subrel r {a | p a}) :=
-  instIsWellOrderElem _ _
-
-instance (r : α → α → Prop) [IsRefl α r] (p : Set α) : IsRefl p (Subrel r p) :=
+instance (r : α → α → Prop) [IsRefl α r] (p : α → Prop) : IsRefl _ (Subrel r p) :=
   ⟨fun x => @IsRefl.refl α r _ x⟩
 
-instance (r : α → α → Prop) [IsSymm α r] (p : Set α) : IsSymm p (Subrel r p) :=
+instance (r : α → α → Prop) [IsSymm α r] (p : α → Prop) : IsSymm _ (Subrel r p) :=
   ⟨fun x y => @IsSymm.symm α r _ x y⟩
 
-instance (r : α → α → Prop) [IsAsymm α r] (p : Set α) : IsAsymm p (Subrel r p) :=
+instance (r : α → α → Prop) [IsAsymm α r] (p : α → Prop) : IsAsymm _ (Subrel r p) :=
   ⟨fun x y => @IsAsymm.asymm α r _ x y⟩
 
-instance (r : α → α → Prop) [IsTrans α r] (p : Set α) : IsTrans p (Subrel r p) :=
+instance (r : α → α → Prop) [IsTrans α r] (p : α → Prop) : IsTrans _ (Subrel r p) :=
   ⟨fun x y z => @IsTrans.trans α r _ x y z⟩
 
-instance (r : α → α → Prop) [IsIrrefl α r] (p : Set α) : IsIrrefl p (Subrel r p) :=
+instance (r : α → α → Prop) [IsIrrefl α r] (p : α → Prop) : IsIrrefl _ (Subrel r p) :=
   ⟨fun x => @IsIrrefl.irrefl α r _ x⟩
 
 end Subrel
 
 /-- Restrict the codomain of a relation embedding. -/
-def RelEmbedding.codRestrict (p : Set β) (f : r ↪r s) (H : ∀ a, f a ∈ p) : r ↪r Subrel s p :=
+def RelEmbedding.codRestrict (p : Set β) (f : r ↪r s) (H : ∀ a, f a ∈ p) : r ↪r Subrel s (· ∈ p) :=
```

Comment:
```suggestion
def RelEmbedding.codRestrict (p : β→ Prop) (f : r ↪r s) (H : ∀ a, p (f a)) : r ↪r Subrel s p :=
```

Reply:
I don't agree with this change, as [`Set.codRestrict`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Set/Function.html#Set.codRestrict) is written in terms of a `Set`.